sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -66,8 +68,17 @@ export class ApplicationController extends BaseController {
     @Body() payload: CreateApplicationDto,
     @UserToken() userToken: IUserToken,
   ): Promise<number> {
+    const programYear = await this.programYearService.validateProgramYear(
+      payload.selectedForm,
+    );
+
+    if (!programYear) {
+      throw new BadRequestException(
+        "Program Year is not active. Not able to create an application invalid request",
+      );
+    }
     const submissionResult = await this.formService.dryRunSubmission(
-      "SFAA2022-23",
+      payload.selectedForm,
```

Comment:
Define the program year based on the form name is only valid if the column "programYear.formName" is unique in the database. IMO, or the column should be unique or we should send the program year id in the payload instead of the form name.

Reply:
We are sending the form name, as its not changed inside the form, please verify sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue. We are dynamically loading the form from the db data, using the drop down. Its better to have the selected form(as it is not tampered in anyway), rather than having the program Year.

Reply:
That was not what we talked about and agreed on the planning/grooming. I am aware that we have the form name, but we talked about it and we decided to have the program id. That's why the AC was defined as below.
What does it means "as it is not tampered in anyway"?

![image](https://user-images.githubusercontent.com/61259237/129603508-c7068b1f-2412-4f5e-950d-b29a6ce335d7.png)

Reply:
We are saving the program year id, but we are using the selected form to retrieve it. Yeah Submission and not in draft state.

Reply:
The sentence mentions the application submission in the sense of the HTTP request being submitted to the API, not the application being saved as submitted. Besides that, we talked about it, and we talked in different meetings. If there is a team decision to keep it as is, not a problem.

Reply:
I dont think, its the HTTP request submission. Then we would have clearly mentioned it as draft and submitted state submissions. Let's discuss